src/test/java/org/drugis/trialverse/dataset/controller/DatasetControllerTest.java

```diff
@@ -362,4 +364,21 @@ public void testGetSpecificVersionInfo() throws Exception {
     verify(historyService).getVersionInfo(datasetUri, versionUri);
   }
 
+  @Test
+  public void testSetArchivedStatus() throws Exception {
+    String datasetUuid = "someDataset";
```

Comment:
should also test the branch in the controller when john is not allowed to do this